microservices-shared-database/src/main/java/com/iluwater/microservices/shared/database/App.java

```diff
@@ -0,0 +1,23 @@
+package com.iluwater.microservices.shared.database;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
+
+/**
+ * Entry point for the shared database microservice application.
+ * This application is configured to exclude the default DataSource auto-configuration as
+ * it is using a custom database configuration.
+ */
```

Comment:
Explain the pattern briefly here. Then describe how the example code implements it.